src/cljs/rems/flash_message.cljs

```diff
@@ -1,91 +1,97 @@
 (ns rems.flash-message
-  (:require [clojure.test :refer [deftest is testing]]
-            [re-frame.core :as rf]
+  (:require [clojure.test :refer [deftest is]]
             [reagent.core :as reagent]
+            [re-frame.core :as rf]
             [rems.administration.status-flags :as status-flags]
             [rems.atoms :as atoms]
             [rems.focus :as focus]
             [rems.text :refer [text text-format]]))
 
-(rf/reg-sub ::message (fn [db _]
-                        (-> (::message db)
-                            (assoc ::message-id (::message-id db)))))
+(defn- location-to-id [location]
+  (let [_ (assert (keyword? location))]
+    (str "flash-message-" (name location))))
 
-(rf/reg-event-fx
- ::reset
- (fn [{:keys [db]} _]
-   {:db (dissoc db ::message)}))
+(deftest test-location-to-id
+  (is (= "flash-message-top" (location-to-id :top))))
 
 (defn- current-time-millis []
   (.getTime (js/Date.)))
 
 (defn- expired? [message]
-  (let [expires (get message :expires 0)]
-    (< expires (current-time-millis))))
+  (< (:expires message 0) (current-time-millis)))
+
+(rf/reg-sub ::message
+            (fn [db [_ location]]
+              (get-in db [::message location])))
+(rf/reg-event-fx ::reset
+                 (fn [{:keys [db]} [_ location]]
+                   {:db (if (some? location)
+                          (update db ::message #(dissoc % location))
+                          (dissoc db ::message))}))
+(rf/reg-event-fx ::show-flash-message
+                 (fn [{:keys [db]} [_ message opts]]
+                   (focus/focus-element-async (str "#" (location-to-id (:location message))))
+                   (when-some [timeout (:timeout opts)]
+                     (js/setTimeout #(rf/dispatch [::reset (:location message)]) timeout))
```

Comment:
could be unnecessary feature as timeout is not used - `:expires` in this flash message component has a different purpose (which is now documented).